emcc.py

```diff
@@ -3307,7 +3307,14 @@ def phase_embind_emit_tsd(options, in_wasm, wasm_target, memfile, js_syms):
   # The Wasm outfile may be modified by emscripten.run, so use a temporary file.
   outfile_wasm = in_temp('tsgen_a.out.wasm')
   emscripten.run(in_wasm, outfile_wasm, outfile_js, memfile, js_syms)
-  out = shared.run_js_tool(outfile_js, [], stdout=PIPE)
+  # Build the flags needed by Node.js to properly run the output file.
+  node_args = []
+  if settings.MEMORY64:
+    node_args += ['--experimental-wasm-memory64']
```

Comment:
We have `shared.node_memory64_flags()` for this.

We don't have the same thing for wasm-eh since it looks like we only allow that to run on node 17 and above: https://github.com/emscripten-core/emscripten/blob/aa250479f654fb8dffb0ff69132da1779128da99/test/common.py#L695-L701.   This suggests to me that the wasm-eh support in node 16 doesn't work, even with the `--experimental-wasm-eh`.   @aheejin do you know if this is true?

Reply:
@sbc100 updated the code to use the shared helper. As for the exception handling, given the fact that it shouldn't have any impact in the generated typings (correct?), would it make sense to just disable it as done with other options?

Reply:
Assuming exceptions can be disabled at link time like this then yes that sounds good.

Reply:
I did this change and the tests seem to pass ✅

Reply:
> We have `shared.node_memory64_flags()` for this.
> 
> We don't have the same thing for wasm-eh since it looks like we only allow that to run on node 17 and above:
> 
> https://github.com/emscripten-core/emscripten/blob/aa250479f654fb8dffb0ff69132da1779128da99/test/common.py#L695-L701
> 
> . This suggests to me that the wasm-eh support in node 16 doesn't work, even with the `--experimental-wasm-eh`. @aheejin do you know if this is true?

Sorry for the delayed reply. #18654 says v16 didn't support the newly added `{traceStack}` option used for stack traces.

Reply:
Ok then, do you think it would be safer to return an error when trying to generate typings with a Node.js version that doesn't properly support wasm exceptions (i.e. < 17)?